libyara/ahocorasick.c

```diff
@@ -871,16 +1618,27 @@ int yr_ac_add_string(
 
     for (int i = 0; i < atom->atom.length; i++)
     {
-      YR_AC_STATE* next_state = _yr_ac_next_state(state, atom->atom.bytes[i]);
+      if (atom->atom.mask[i] == YR_ATOM_TYPE_ANY)
+      {
+        yr_bitmask_clear_all(atom->atom.bitmap[i], sizeof(atom->atom.bitmap[i]));
+        for (int j = 0; j < YR_BITMAP_SIZE; j++)
```

Comment:
The intention here is setting all the bits in the bitmap to 1, so implementing `yr_bitmask_set_all` using `memset` would be clearer and more efficient.